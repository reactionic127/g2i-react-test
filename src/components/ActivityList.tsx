import React, { ReactElement } from "react";
import { List, Typography, Avatar } from "antd";
import { ListType } from "../types";

const currentUser = {
  name: "Richard Gillbert",
  shortname: "RG",
};

function ActivityList({ activities }: ListType): ReactElement {
  return (
    <List
      data-id="list"
      style={{ marginTop: 20, maxHeight: 300, overflowY: "auto" }}
      bordered
      dataSource={activities}
      renderItem={(item) => {
        return (
          <List.Item data-id="list-item">
            <Avatar
              style={{ backgroundColor: "#F0685E", marginRight: 10 }}
              size={40}
            >
              {currentUser.shortname}
            </Avatar>
            <Typography.Text strong>
              {currentUser.name} added a {item.type}:&nbsp;
            </Typography.Text>
            {item.url} {item.note}
          </List.Item>
        );
      }}
    />
  );
}

export default ActivityList;
