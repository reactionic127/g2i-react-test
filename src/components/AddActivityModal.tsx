import React, { ReactElement, useState } from "react";
import { useDispatch } from "react-redux";
import { Input, Modal, Select, Space } from "antd";
import { ModalType, SelectType } from "../types";
import { ADD_ACTIVITY_REQUEST } from "../redux/constants";

const { Option } = Select;
const { TextArea } = Input;

function AddActivityModal({ visible, setVisible }: ModalType): ReactElement {
  const dispatch = useDispatch();

  const [type, setType] = useState<string>("note");
  const [url, setUrl] = useState<string>("");
  const [note, setNote] = useState<string>("");

  const handleChange = (value: SelectType) => {
    setType(value.value);
  };

  const submitValues = () => {
    if (note) {
      dispatch({ type: ADD_ACTIVITY_REQUEST, payload: { type, url, note } });
      setVisible(false);
      setNote("");
      setUrl("");
    }
  };

  return (
    <Modal
      data-id="modal-container"
      title="Add Activity"
      centered
      visible={visible}
      okText="Submit"
      onOk={submitValues}
      cancelButtonProps={{ style: { display: "none" } }}
      onCancel={() => setVisible(false)}
    >
      <Space direction="vertical" size="large" style={{ width: "100%" }}>
        <Select
          data-id="select-type"
          labelInValue
          defaultValue={{ value: "note", label: "Add Note", key: "note" }}
          style={{ width: 150 }}
          onChange={handleChange}
        >
          <Option value="note">Add Note</Option>
          <Option value="enquiry">Add Enquiry</Option>
          <Option value="task">Add Task</Option>
        </Select>
        <Input
          data-id="property-input"
          placeholder="Add Property URL (optional)"
          style={{ width: 300 }}
          value={url}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setUrl(e.target.value)
          }
        />
        <TextArea
          data-id="note-input"
          rows={4}
          placeholder="Add notes..."
          style={{ width: "100%", resize: "none" }}
          value={note}
          onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) =>
            setNote(e.target.value)
          }
        />
      </Space>
    </Modal>
  );
}

export default AddActivityModal;
