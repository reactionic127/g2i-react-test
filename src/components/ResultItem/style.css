.result-item {
  display: flex;
  align-items: center;
  margin: 10px 0;
  color: rgb(140, 0, 0);
  padding: 0 20px;
}
.result-item.correct {
  color: green;
}
.result-sign {
  width: 30px;
  text-align: center;
  font-size: 24px;
  margin-right: 10px;
}
.question {
  flex: 1;
  text-align: left;
  word-break: break-word;
  font-size: 18px;
}
