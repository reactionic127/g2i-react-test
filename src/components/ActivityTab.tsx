import React, { ReactElement, useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { PlusCircleFilled } from "@ant-design/icons";
import AddActivityModal from "./AddActivityModal";
import ActivityList from "./ActivityList";
import { GET_ALL_ACTIVITIES_REQUEST } from "../redux/constants";
import getActivityState from "../redux/selectors";

function ActivityTab(): ReactElement {
  const dispatch = useDispatch();
  const { activities } = useSelector(getActivityState);

  const [modalVisible, setModalVisible] = useState(false);

  useEffect(() => {
    dispatch({ type: GET_ALL_ACTIVITIES_REQUEST });
  }, [dispatch]);

  return (
    <div>
      <button
        data-id="add-button"
        type="button"
        className="create-activity-btn"
        onClick={() => setModalVisible(true)}
      >
        <PlusCircleFilled style={{ fontSize: 22, color: "#4994FF" }} />
        <span className="activity-text">Add new activity</span>
      </button>
      <AddActivityModal
        data-id="add-activity-modal"
        visible={modalVisible}
        setVisible={setModalVisible}
      />
      <ActivityList activities={activities} />
    </div>
  );
}

export default ActivityTab;
