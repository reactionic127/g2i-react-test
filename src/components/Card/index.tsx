import cx from "classnames";

import "./style.css";

type Props = {
  title: string;
  handleClick: (args: string) => void;
};

export const QuestionCard = ({ title, handleClick }: Props) => (
  <div className={cx("question-card")}>
    <h2>{title}</h2>
    <button className={cx("btn-answer")} onClick={() => handleClick("True")}>
      TRUE
    </button>
    <button className={cx("btn-answer")} onClick={() => handleClick("False")}>
      FALSE
    </button>
  </div>
);
