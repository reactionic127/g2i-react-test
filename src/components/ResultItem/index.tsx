import cx from "classnames";
import "./style.css";

type Props = {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  question: any;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  answer: any;
};

export const ResultItem = ({ question, answer }: Props) => {
  const isCorrect = question.correct_answer === answer;

  return (
    <div
      className={cx({
        "result-item": true,
        correct: isCorrect,
      })}
    >
      <div className={cx("result-sign")}>{isCorrect ? "+" : "-"}</div>
      <div className={cx("question")}>{question.question}</div>
    </div>
  );
};
