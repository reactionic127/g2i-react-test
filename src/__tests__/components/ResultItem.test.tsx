import { shallow, ShallowWrapper } from "enzyme";
import { ResultItem } from "../../components/ResultItem";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  useSelector: () => ({ activities: [] }),
  useDispatch: () => mockDispatch,
}));

describe("ActivityList", () => {
  describe("rendering", () => {
    let wrapper: ShallowWrapper;
    const mockQuestion = {
      category: "Entertainment: Video Games",
      type: "boolean",
      difficulty: "hard",
      question:
        "In &quot;The Sims&quot; series, the most members in a household you can have is 8.",
      correct_answer: "True",
      incorrect_answers: ["False"],
    };

    beforeEach(() => {
      wrapper = shallow(<ResultItem question={mockQuestion} answer={"True"} />);
    });

    it("renders the component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
