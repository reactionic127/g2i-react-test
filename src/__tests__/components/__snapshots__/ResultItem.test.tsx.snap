// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem rendering renders the component 1`] = `
<div
  className="result-item correct"
>
  <div
    className="result-sign"
  >
    +
  </div>
  <div
    className="question"
  >
    In &quot;The Sims&quot; series, the most members in a household you can have is 8.
  </div>
</div>
`;
